Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGToDBAnalyzer.cs

```diff
@@ -0,0 +1,165 @@
+
+using System;
+using System.IO;
+using BuildXL.Scheduler.Tracing;
+using BuildXL.ToolSupport;
+using BuildXL.Utilities;
+using Google.Protobuf;
+using BuildXL.Engine.Cache;
+using BuildXL.Engine.Cache.KeyValueStores;
+using BuildXL.Engine.Cache.Serialization;
+using BuildXL.Native.IO;
+using System.Diagnostics;
+using BuildXL.Execution.Analyzer.Model;
+using BuildXL.Analyzers.Core.XLGPlusPlus;
+
+namespace BuildXL.Execution.Analyzer
+{
+    internal partial class Args
+    {
+        public Analyzer InitializeXLGToDBAnalyzer()
+        {
+            string outputDirPath = null;
+            foreach (var opt in AnalyzerOptions)
+            {
+                if (opt.Name.Equals("outputDir", StringComparison.OrdinalIgnoreCase) ||
+                   opt.Name.Equals("o", StringComparison.OrdinalIgnoreCase))
+                {
+                    outputDirPath = ParseSingletonPathOption(opt, outputDirPath);
+                }
+                else
+                {
+                    throw Error("Unknown option for event stats analysis: {0}", opt.Name);
+                }
+            }
+
+
+            if (string.IsNullOrEmpty(outputDirPath))
+            {
+                throw Error("/outputDir parameter is required");
+            }
+
+
+            return new XLGToDBAnalyzer(GetAnalysisInput())
+            {
+                OutputDirPath = outputDirPath,
+            };
+        }
+
+        /// <summary>
+        /// Write the help message when the analyzer is invoked with the /help flag
+        /// </summary>
+        private static void WriteXLGToDBHelp(HelpWriter writer)
+        {
+            writer.WriteBanner("XLG to DB \"Analyzer\"");
+            writer.WriteModeOption(nameof(AnalysisMode.XlgToDb), "Dumps event data from the xlg into a database.");
+            writer.WriteOption("outputDir", "Required. The directory to write out the RocksDB database", shortName: "o");
+        }
+    }
+
+
+    /// <summary>
+    /// Analyzer to dump xlg events and other data into RocksDB
+    /// </summary>
+    internal sealed class XLGToDBAnalyzer : Analyzer
+    {
+        public string OutputDirPath;
+        private bool m_accessorSucceeded = true;
+        private BXLInvocationEventList m_bXLInvEveList = new BXLInvocationEventList();
+        private KeyValueStoreAccessor Accessor { get; set; }
+        private uint WorkerID { get; set; }
+
+
+        public XLGToDBAnalyzer(AnalysisInput input) : base(input)
+        {
+           
+        }
+
+        /// <inheritdoc/>
+        public override void Prepare()
+        {
+            try
+            {
+                Directory.Delete(path: OutputDirPath, recursive: true);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("No such dir or could not delete dir with exception {0}.\nIf dir still exists, this analyzer will append data to existing DB.", e);
+            }
+
+            var dataStore = new XLGppDataStore();

```

Comment:
>var dataStore = new XLGppDataStore(); [](start = 12, length = 37)

so you have this `XLGppDataStore` class that handles (?) access to the db. However, here you only use that class to get the accessor. If it's the only thing you need XLGppDataStore here for, should we create the accessor here direclty (ie., without roundtripping through that new class)

Reply:
Im going to go ahead and make the datastore here instead of roundtripping

---
In reply to: [306055600](https://github.com/microsoft/BuildXL/pull/636#discussion_r306055600) [](ancestors = 306055600)